### DISCUSSION TITLE
Feedback needed on new UI - Plug-in Configurations

### DISCUSSION THREAD

**Discussion by scottrenaud** [Autodesk/maya-usd#3616] [Show and tell]
Hey everyone,

We’d like your feedback on some **Plug-in Configuration** designs!

Here are the proposed enhancements:

- Artists can select one or more plug-in configurations at a time when importing or exporting
- Plugin writers can hook up a custom UI that can expose options for artists to configure

**Plug-in configurations on export** (wireframe)

https://github.com/Autodesk/maya-usd/assets/108303538/ba1e0a36-694c-46b9-8b9b-c1c9cb89a5e4

We'd like to know:
- Do you have any questions or concerns?
- Is there anything you would you do differently?
- How many people at your studio are writing plug-in configurations (also known as job contexts)?
- How many plug-in configurations can you see your artists having in their list in the UI?

Don’t feel like answering here?  
Here’s a [survey link](https://autodeskfeedback.az1.qualtrics.com/jfe/form/SV_cSbUy0yhp1W7HaS) to respond anonymously.

Thanks! 
Scott 
[scott.renaud@autodesk.com](mailto:scott.renaud@autodesk.com)

**Comment by swillisart** [comment]
Hey Scott!
The options are far too spaced out and margins are consuming too much space in an already small window. 
A large number of users (artists) use tablets and the consistent over-spacing of the UI is forcing an unnecessary amount of scrolling across the board. Please escalate this up the chain.
 
I like that they are each using their own option windows.

**Reply by scottrenaud** [reply]
Good point, thanks @swillisart!

**Comment by sharktacos** [comment]
I'd like to see more options in the Arnold "Options" dialog. Specifically, I'd like to be able to export materials and their assignments/bindings without the geometry.

**Reply by scottrenaud** [reply]
Hey @sharktacos, good news we're actually working on designs for exporting materials only.
As for the bindings themselves, would you have some example scenarios for when folks would want to do this and how they'd be leveraged?

**Reply by sharktacos** [reply]
Our current workflow is to use [this script](https://github.com/sharktacos/VFX-software-prefs/blob/main/Maya/scripts/df_USD_mtlExport.py) which uses the current Maya USD export and then parses that file removing the geo from it, leaving only the materials and their bindings. We then layer that "mtl.usd" over the "geo.usd" file in order to keep the work of the modeling and lookdev departments independent. I have a video outlining that process here: https://youtu.be/mIGvq5-7HAI?si=Sm75Qk8iZ1ZmTmea

**Reply by scottrenaud** [reply]
Hey @sharktacos, thanks for the awesome context!
I just created a discussion on [exporting materials](https://github.com/Autodesk/maya-usd/discussions/3674) to get feedback on some designs.  It doesn't include exporting bindings on their own...yet, but it's a start.  Would love to get your feedback, thanks!
